src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -565,14 +555,7 @@ def call(
         return_dict=None,
         training=False,
     ):
-        output_attentions = output_attentions if output_attentions is not None else self.output_attentions
-        output_hidden_states = output_hidden_states if output_hidden_states is not None else self.output_hidden_states
-        use_cache = use_cache if use_cache is not None else self.use_cache
-        return_dict = return_dict if return_dict is not None else self.config.return_dict
-        if use_cache:
-            assert not training, "Training + use cache are incompatible"
         # check attention mask and invert
-        use_cache = cast_bool_to_primitive(use_cache)
```

Comment:
Does removing this mean the `TFBartDecoder` will not be able to handle the output attentions/hidden states/cache/return dict parameters on its own, but will instead rely on being called by the class `TFBart{Model, ForConditionalGeneration}`?

If @patrickvonplaten is doing the same refactor for TF BART as he did for the PT version, then one of the enhancements it offers is being able to use the `(TF)BartDecoder` as a standalone model. Will this change prevent this?

Reply:
If the `(TF)BartDecoder` aims to be used as a standalone, the `input_processing` must be added. I can add it just in case, it won't arm the current bebavior, just an "over" processing.  Should I do it?

Reply:
I think @patrickvonplaten will probably take care of that in his PR, so no problem here. I would still wait for his review before merging!

Reply:
I haven't looked at having neither TFBart nor TFT5 as "encoder-only" or "decoder-only" model yet because a) there is not TFEncoderDecoder model and b) because of the issues that will be solved in this PR. So I'm 100% fine to delete it for now here